pandas/io/json/json.py

```diff
@@ -81,14 +87,15 @@ def __init__(self, obj, orient, date_format, double_precision,
         self.ensure_ascii = ensure_ascii
         self.date_unit = date_unit
         self.default_handler = default_handler
+        self.index = index
 
         self.is_copy = None
         self._format_axes()
 
     def _format_axes(self):
         raise AbstractMethodError(self)
 
-    def write(self):
+    def _write(self):
```

Comment:
not really what I meant

``.write()`` should remain in the superclass as the public interface with no arguments, ``_write(obj)`` will take an ``obj`` for example, so we dont' mutate in the subclasses (you will instead pass the revised obj in the super call)

Reply:
So should there be a ``.write(self)`` and ``._write(self, obj)`` method in the ``Writer`` class and then a ``._write(self, obj)`` method in each of the subclasses (``SeriesWriter``, ``FrameWriter`` and ``JSONTableWriter``)? 

Then the ``._write(self, obj)`` methods in each of the subclasses call ``super()._write(obj)`` after making any necessary adjustments to ``obj``? 

And in the ``to_json()`` function at the top of the file we use ``._write(obj)`` on an instance of the relevant class?

Reply:
.write() will just call ._write()

so the original api is preserved